cdap-docs/developers-manual/source/building-blocks/workflows.rst

```diff
@@ -274,11 +385,104 @@ otherwise, the other path will be taken. The diagram for this code would be:
    :width: 8in
    :align: center
 
-For this 3.0 release, CDAP only supports predicate conditions based on counters from
-previous MapReduce programs. Because of this, condition nodes are currently only useful
-when they follow MapReduce nodes; they can then make use of counters emitted by the
-MapReduce program to take the decision. In a later version, we will add more information
-to the ``WorkflowContext`` which will allow for more complex decisions.
+Workflow Token with Forks and Joins
+-----------------------------------
+For workflows that involve forks and joins, a single instance of the workflow token is
+shared by all branches of the fork. Updates to the singleton are made thread-safe through
+synchronized updates, guaranteeing that value you obtain from reading the token is the
+last value written.
+
+Example
+-------
+
+This code sample shows how to obtain values from the token from within a custom action,
+and from within a workflow with a predicate, fork and joins::
+
+  @Override
+  public void run() {
+    ...
+    WorkflowToken token = getContext().getToken();
+    
+    // Set the type of action of the current node:
+    token.put("action_type", "CUSTOM_ACTION");
```

Comment:
Hmmm, we should not allow updates to workflow token in ``run()``, since we do not allow it in ``map``, ``reduce``. Perhaps override the ``initialize(WorkflowContext context)`` method in this example, and update it there.